File: pykatas/count_pairs.py
from typing import List


class Solution:
    def count_pairs(self, nums: List[int], target: int) -> int:
        left, right = 0, 1
        result = 0

        while left < right:
            if right > len(nums) - 1:
                left += 1
                right = left + 1

                if right >= len(nums):
                    break

            if nums[left] + nums[right] < target:
                result += 1

            right += 1

        return result


sol = Solution()
nums = [-1, 1, 2, 3, 1]
target = 2
print(sol.count_pairs(nums, target))

File: gokatas/countpairs.go
package gokatas

// Another solution would be two for loops i and j
// where a inner loop for j starts in the position 1 in the array
func countPair(nums []int, target int) int {
	left, right := 0, 1
	result := 0

	for left < right {
		if right > len(nums)-1 {
			left++
			right = left + 1

			if right >= len(nums) {
				break
			}
		}

		if nums[left]+nums[right] < target {
			result++
		}

		right++
	}

	return result
}
